fedora_messaging/tests/unit/test_session.py

```diff
@@ -515,6 +540,194 @@ def test_connection_error_with_pika_exception_as_error(self):
+        mock_connection = mock.Mock(**conn_mock_attrs)
+        self.consumer._connection = mock_connection
+        self.consumer._channel = mock.Mock()
+        reply_code_or_reason = "reply_code"
+        reply_text = "reply_text"
+        with mock.patch("fedora_messaging._session._log") as mock_log:
+            self.consumer._on_connection_close(
+                mock_connection, reply_code_or_reason, reply_text
+            )
+        mock_log.warning.assert_called_once_with(
+            "Connection to %s closed unexpectedly (%d): %s",
+            host_name,
+            0,
+            reply_code_or_reason,
+        )
+        self.assertEqual(self.consumer._channel, None)
+        mock_connection.ioloop.call_later.assert_called_once_with(
+            1, self.consumer.reconnect
+        )
+
+    def test_signal_SIGTERM(self):
+        """Assert ConsumerSession._shutdown is called on signal SIGTERM."""
+        consumer_tags = "test_consumer_tags"
+        channel_mock_attrs = {"consumer_tags": consumer_tags}
+        mock_channel = mock.Mock(**channel_mock_attrs)
+        conn_mock_attrs = {"is_open": True}
+        self.consumer._running = True
+        mock_connection = mock.Mock(**conn_mock_attrs)
+        self.consumer._connection = mock_connection
+        self.consumer._channel = mock_channel
+        signal_handler = signal.getsignal(signal.SIGTERM)
+
+        with mock.patch("fedora_messaging._session._log") as mock_log:
+            signal_handler(signal.SIGTERM, "test_frame")
+
+        mock_log.info.assert_called_once_with(
+            "Halting %r consumer sessions", consumer_tags
+        )
+        self.assertEqual(self.consumer._running, False)
+        mock_connection.close.assert_called_once_with()
+        self.assertEqual(signal.getsignal(signal.SIGTERM), signal.SIG_DFL)
+
+    def test_signal_SIGINT(self):
+        """Assert ConsumerSession._shutdown is called on signal SIGINT."""
+        consumer_tags = "test_consumer_tags"
+        channel_mock_attrs = {"consumer_tags": consumer_tags}
+        mock_channel = mock.Mock(**channel_mock_attrs)
+        conn_mock_attrs = {"is_open": True}
+        mock_connection = mock.Mock(**conn_mock_attrs)
+        self.consumer._running = True
+        self.consumer._connection = mock_connection
+        self.consumer._channel = mock_channel
+        signal_handler = signal.getsignal(signal.SIGINT)
+
+        with mock.patch("fedora_messaging._session._log") as mock_log:
+            signal_handler(signal.SIGINT, "test_frame")
+
+        mock_log.info.assert_called_once_with(
+            "Halting %r consumer sessions", consumer_tags
+        )
+        self.assertEqual(self.consumer._running, False)
+        mock_connection.close.assert_called_once_with()
+        self.assertEqual(signal.getsignal(signal.SIGINT), signal.SIG_DFL)
+
+    def test_signal_SIGWINCH(self):
+        """Assert ConsumerSession._shutdown is called on signal SIGIN."""
```

Comment:
I assume this is supposed to say "Assert ConsumerSession._shutdown isn't called on SIGWINCH."

I think this could be replaced with a test that iterates through all the defined signals except SIGINT and SIGTERM and asserts that they are set to the default handler.